int32 id
sensor_msgs/Image spec
string result